src/main.ts

```diff
@@ -21,4 +49,133 @@ async function run() {
   }
 }
 
+async function handleMain(inputs: Inputs, client: GitHubClient): Promise<void> {
+  const changes = await getChangesInPush();
+
+  if (changes.additions.includes(inputs.haltFile)) {
+    core.startGroup(`${inputs.defaultBranch}:${inputs.haltFile} added`);
+    const message = haltMessage(fs.readFileSync(inputs.haltFile).toString());
+    core.info(`Halting all open PRs: ${message}`);
+    await haltOpenPullRequests(client, github.context, inputs, message);
+    core.endGroup();
+  }
+
+  if (changes.removals.includes(inputs.haltFile)) {
+    core.startGroup(`${inputs.defaultBranch}:${inputs.haltFile} removed`);
+    core.info("Un-halting all open PRs");
+    await unhaltOpenPullRequests(client, github.context, inputs);
+    core.endGroup();
+  }
+}
+
+async function handlePullRequest(
+  inputs: Inputs,
+  client: GitHubClient,
+  pullRequest: PullRequest
+): Promise<void> {
+  const haltFile = await githubApi.getRepositoryContent(client, {
+    ...github.context.repo,
+    path: inputs.haltFile,
+    ref: inputs.defaultBranch,
+  });
+
+  const haltFileContents =
+    haltFile && "content" in haltFile ? atob(haltFile.content) : null;
+
+  if (!haltFileContents) {
```

Comment:
Will `atob` ever return an falsy value? For example 

```js
"" == false
-- true
```

That would make an empty file a green repo.

Reply:
Yup, good catch. That's annoying,

```console
> if (!"") { console.log("Empty string is falsey"); }
Empty string is falsey
undefined
>
```

I'll make this the stricter `!== null` I guess -- is that the idiomatic way?

Reply:
Couple of things,

Technically, we'd be OK here. An "empty" halt file would still be the string `"\n"` with it's final trailing newline (unless someone worked really hard to avoid it -- although I know you have a knack). I'll still fix this, but just FYI.

`haltFileContents !== null` doesn't appease TypeScript -- it won't change the type from `string | null` into `string` the same way `!haltFileContents` does. So that's fun.

So it seems `(!haltFileContents && haltFileContents !== "")` is what I actually need. Such elegance.

Reply:
Scratch that -- I was mixed up. Sorry TypeScript, you're good.